### ISSUE TITLE
Left padding makes prefix masking incorrect in SFT collator (labels alignment bug)

### ISSUE THREAD

**Issue by Acecy237** [QwenLM/Qwen3-ASR#70]
### Description

**问题描述**
当前 `finetuning/qwen3_asr_sft.py` 的 `DataCollatorForQwen3ASRFinetuning` 在构造 `labels` 时，用 `prefix_lens` 直接从左侧开始 mask。  
但 `processor` 使用左 padding，`attention_mask.sum()` 只统计非 pad 长度，导致 mask 位置偏移，prefix 真实 token 未被正确遮挡。

### Reproduction

相关代码：
  ```
  for i, pl in enumerate(prefix_lens):
      start = seq_len - full_lens[i]
      labels[i, start : start + pl] = -100
  ```

### Logs

```shell

```

### Environment Information

latest code

### Known Issue

- [x] The issue hasn't been already addressed in Documentation, Issues, and Discussions.

**Comment by Acecy237** [comment]
[debug] tokenizer.padding_side = right
[debug] prefix_attention_mask[0][:32] = [0, 0, 0, ... , 1, 1]
[debug] full_attention_mask[0][:32] = [0, 0, 0, ... , 1, 1]

**Comment by binnybn98** [comment]
This prefix masking incorrect issue will lead to a "!" output when inference by a finetuned model?

**Comment by Acecy237** [comment]
@binnybn98 idk, I have fixed it  before finetuned.

**Comment by binnybn98** [comment]
@Acecy237 How did you fix it? Could you elaborate more or paste your fixed code up here? Cause when I follow the official finetune guide and do inference via finetuned model, my output is one "!"

**Comment by lq0104** [comment]
> Reproduction
> 相关代码：
> 
> ```
> for i, pl in enumerate(prefix_lens):
>     start = seq_len - full_lens[i]
>     labels[i, start : start + pl] = -100
> ```

想请问一下@Acecy237 ，你这个代码是在哪看到的还是说这是你修改后的代码，因为我看原始的代码是以下这样的，没有看到你贴的代码呢？
https://github.com/QwenLM/Qwen3-ASR/blob/c17a131fe028b2e428b6e80a33d30bb4fa57b8df/finetuning/qwen3_asr_sft.py#L145-L146
